### ISSUE TITLE
12.1 ALPHA - EmailForm Block No Longer Works

### ISSUE THREAD

**Issue by JimMichael** [SparkDevNetwork/Rock#4582]
### Description

This is related to https://github.com/SparkDevNetwork/Rock/issues/4550 but I'm entering it as new since this is new behavior as of 12.1 alpha. 

The "fixed in 12.1" for the issue linked above **did not work**. Now the EmailForm block does not work at all. It no longer throws an exception, but also no longer delivers any email.

### Steps to Reproduce

1. Put EmailForm block on a page
2. Configure appropriate to/from addresses.
3. Send the form
4. Observe the email never arrives

**Expected behavior:**

EmailForm block would work as it did prior to 12.0

**Actual behavior:**

It doesn't work at all, now.  Note that because the exception it was causing in v12 was related to the "No Legacy" global attribute setting, I did try it with "Legacy with warning" instead, but it still did not deliver.

### Versions

* **Rock Version:** 12.1

**Comment by mikedotmundy** [comment]
This does work for me on 12.1. I had no issues and it did NOT cause any exceptions. I use Mailgun SMTP Transport.

**Comment by JimMichael** [comment]
After further testing, I've found that it only seems to be a problem when we're using the _SendGrid HTTP_ transport. I tried some others (all pointing at our SendGrid account) and here's the results:

SendGrid HTTP: NO email arrives
SendGrid SMTP (old plugin): email arrives
SMTP: email arrives.

NO Exceptions are thrown in any of the three scenarios. Hopefully that helps narrow down the issue.

**Comment by matthewbronkema** [comment]
@JimMichael can you enable the Save Communication History on the email form and then retry your SendGrid HTTP test? I am hoping the results saved in the communication will give us some more troubleshooting information.

**Comment by JimMichael** [comment]
@matthewbronkema Enabling Save Communication History causes the email _to be delivered successfully_. With it disabled again, the mail never arrives.

After further testing, I've also discovered that this is not isolated to the EmailForm block. We were also seeing "random" workflows that use Send Email action failing (but again, only when using SendGrid HTTP transport). I looked at a few of these just now, and sure enough... the ones that work already have Save Communication History enabled in the action, and the ones that fail to deliver, don't.

So there's definitely something wonky going on with SendGrid HTTP transport when save comm history is disabled.
